Mds 125

ajchilts opened the pull request:
**IMPORTANT:** - Some HO fields default to zero where a value is not present in the DES response. I have preserved this behaviour for MOJ. This needs to be addressed to see if this behaviour occurs in the other APIs? This could present a problem if we need to default to zero. I.e. We can't just default to zero if not present (as in HO) as it may not be present as it is not in scope!! I think this is less of an issue for SA (Income) endpoints as the individual APIs are themselves scoped. It is therefore safe to assume that the value is not present; has been asked for (as we are able to hit this endpoint) and should default to zero. 

cc @Bongowhack @Stu-Price @owen-opencast ^^

**Postman testing guide:**

Stub: https://github.com/hmrc/if-stub/tree/MDS-125

In a browser

Visit http://localhost:9949/auth-login-stub/gg-sign-in
Redirect URL: http://localhost:9949/auth-login-stub/session
Enrolment Keys:
read:individuals-matching
_read:individuals-income-laa-c1_ (important so that fields query matches payload in IF) **OR**
_read:individuals-income-lsani-c1_

Submit and copy authToken

Using Postman

Get matchId

POST http://localhost:9652
Authorization: Bearer token (paste in authToken from above)
Headers:
Content-Type: application/json
Accept: application/vnd.hmrc.P1.0+json
Body:
{
"firstName": "John",
"lastName": "Densmore",
"nino": "CS700100A",
"dateOfBirth": "1969-03-05"
}

Copy matchId from:
“/individuals/matching/”

Income Root

GET http://localhost:9652/
Authorization: Bearer token (paste in authToken from above)
Headers:
Accept: application/vnd.hmrc.P2.0+json
Query Parameters:
matchId:

Income SA

GET http://localhost:9652/sa (ALL)
**NOTE: HIT ROUTE FIRST TO GET SCOPE RESTRICTED LIST OF AVAILABLE ENDPOINTS**
Authorization: Bearer token (paste in authToken from above)
Headers:
Accept: application/vnd.hmrc.P2.0+json

Query parameters:

_matchId: <from matching>
fromTaxYear=2017-18
toTaxYear=2018-19_

owen-opencast commented on test/unit/uk/gov/hmrc/individualsincomeapi/domain/v2/SaFurtherDetailsSpec.scala:
```diff
@@ -0,0 +1,120 @@
+  val ifSaNoData = Seq(createValidSaTaxYearEntryNoDataContainers())
+  val ifSaNoValues = Seq(createValidSaTaxYearEntryNoValues())
+
+  // TODO Q. Do we need the items in this json response to default to zero as other SA responses do?
```
TODO

ajchilts replied:
I want to leave this question open until I have discussed with @Bongowhack on his return. 👍

owen-opencast commented on app/uk/gov/hmrc/individualsincomeapi/domain/v2/SaPartnerships.scala:
```diff
@@ -0,0 +1,60 @@
+
+  private def default = SaPartnership(0.0)
+
+  private def TransformSaPartnership(entry: IfSaEntry) =
```
Capitalised "Transform" method names in domain objects?

owen-opencast reviewed (changes requested):
Looks good! A few tweaks and queries

ajchilts commented:
@owen-opencast - Addressed all review comments. The 2020 copyright comments have also been changed to 2021 so it looks like more changes when in fact its simply changes to comments.

owen-opencast commented:
Looks good - I think all of the domain objects had capitalised Transform methods, still a few to do 🙂
